### PR TITLE
Test 10

### PR DESCRIPTION
**Pull request by wfeldt**
This is about feature #316245 (simplify bootloader handling). The intention is to get rid of the
overly complex perl-Bootloader code.

For this we switch to simple shell scripts located in /usr/lib/bootloader/<LOADER_NAME>. Currently
grub2(-efi) is implemented (more could follow). This implements just enough to keep update-bootloader
as used in the kernel rpm post-install scripts running.

For backward compatibility the old code is still used if no scripts for the selected bootloader are found.

yast2-bootloader will be changed not to use perl-Bootloader as backend. So perl-Bootloader will only
be used for kernel package updates (via update-bootloader) in future.

### CODE REVIEW

File: grub2/install [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  else
+    echo "/etc/default/grub_installdevice: no such file"
+    err=1
+  fi
+else
+  echo "grub2-install: command not found"
+  err=1
+fi
+
+exit $err
```
Comment (jreidinger) [inline grub2/install]: does we need it at all? I think that kernel update does not need it.

File: new_pbl.txt [hunk 2]
```diff
@@ -0,0 +1,58 @@
+===  perl-Bootloader - redone  ===
+
+The script we are talking about is here:
+
+https://github.com/openSUSE/perl-bootloader/blob/test_10/pbl
```
Comment (jreidinger) [inline new_pbl.txt]: link stop existing when branch was deleted

File: grub2/install (hunk 1 above)
Comment (wfeldt) [inline grub2/install]: ATM it is used. We might drop it when yast2 has been rewritten.

File: pbl [hunk 3]
```diff
@@ -0,0 +1,224 @@
+#! /usr/bin/perl
```
Comment (jreidinger) [inline pbl]: I think idea was no perl, don't?

File: new_pbl.txt (hunk 2 above)
Comment (wfeldt) [inline new_pbl.txt]: a new link does not yet exist

File: pbl (hunk 3 above)
Comment (wfeldt) [inline pbl]: For the moment it does do. The rest is kept for compat as well. But The script can be rewritten in bash later.

File: grub2/install (hunk 1 above)
Comment (wfeldt) [inline grub2/install]: update-bootloader still has the --reinit option

File: pbl [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+#
+# level: 0 .. 3 (debug, info, warning, error)
+# message: log message (single line string)
+# var (optional): either a SCALAR or a REF
+#   - SCALAR (may be multiline) will be logged in a block delimited
+#     by '<<<<<<<<<<<<<<<<' and '>>>>>>>>>>>>>>>>'
+#   - REF will be logged using Data::Dumper
+# depth (optional): maximum depth when logging a REF
+#
+sub info
```
Comment (jreidinger) [inline pbl]: I think `log` is better name as info indicate log with level info.

Comment (wfeldt) [inline pbl]: log is perl function
I wouldn't really put too much thought into a function name that's called a handfull of times

File: grub2/install (hunk 1 above)
Comment (jreidinger) [inline grub2/install]: yeah, it was old option for lilo, which need reinit everytime its config changed. I think now only possible user is grub2 itself, when it need update stage1 code.

File: pbl (hunk 4 above)
Comment (jreidinger) [inline pbl]: ok, fine

**Comment by jreidinger** [comment]
in general LGTM

**Pull request merged**